code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -71,98 +79,67 @@
 	..()
 	can_hold = typecacheof(can_hold)
 
-/obj/item/gripper/attack_self(mob/user as mob)
-	if(wrapped)
-		wrapped.attack_self(user)
-
-/obj/item/gripper/verb/drop_item()
-
-	set name = "Drop Gripped Item"
-	set desc = "Release an item from your magnetic gripper."
-	set category = "Drone"
-
-	drop_item_p()
-
-// The "p" stands for proc, since I was having annoying weird stuff happening with this in the verb
-// when trying to have default values for arguments and stuff
-/obj/item/gripper/proc/drop_item_p(var/silent = 0)
-
-	if(!wrapped)
-		//There's some weirdness with items being lost inside the arm. Trying to fix all cases. ~Z
-		for(var/obj/item/thing in src.contents)
-			thing.forceMove(get_turf(src))
-		return
-
-	if(wrapped.loc != src)
-		wrapped = null
-		return
+/obj/item/gripper/attack_self(mob/user)
+	if(gripped_item)
+		if(gripped_item.type in configurable_items)
+			var/choice = alert(user, "Do you want to configure/place the item, or drop it?", "Gripper selection", "Configure/place", "Drop", "Cancel")
+			switch(choice)
+				if("Configure/place")
+					gripped_item.attack_self(user)
+				if("Drop")
+					drop_gripped_item()
+				if("Cancel")
+					return FALSE
+		else
+			drop_gripped_item()
+	else
+		to_chat(user, "<span class='warning'>Your gripper is empty.</span>")
 
+/obj/item/gripper/proc/drop_gripped_item(silent = 0)
```

Comment:
There needs to be some check in either here or the `attack_self` to prevent people being able to teleport gripped items. (they can activate the `attack_self()` multiple times)

Reply:
I'm not sure I know what you mean. How can items be teleported?

Reply:
Actually wait never mind, I saw you set `gripped_item` to null